### DISCUSSION TITLE
Right click and drag to zoom on area

### DISCUSSION THREAD

**Discussion by sellicott** [Lramseyer/vaporview#6] [Ideas]
Select horizontal zoom area by click and dragging with the right mouse button (this is a feature in GTKWave).

**Comment by Lramseyer** [comment]
Thanks for the suggestion! It is definitely something I have considered, and will probably get around to doing. I was putting it off because the viewer was acting weird with some zoom ratios and chunk sizes. But those underlying causes have been _mostly_ fixed now. The only hold out is the function I use to draw the ruler. I'm in the process of modifying the zoom and chunk sizing code, so it might be in the cards.

If anyone wants to get their hands dirty and try to implement this feature themselves, I'm happy to do a more in depth write up or even a zoom session.

**Comment by Lramseyer** [comment]
@sellicott - I added this feature last night, and will include it in the 0.9.8 release. I'll post .vsix file tonight with the release notes. If you don't want to wait, you can do a git pull and build it yourself!

The only thing is that the click and drag is with the left mouse button instead (similar to Verdi, which I have more experience with.) I did this because it was easier to attach to the existing event handlers. I also put in some checks so that it doesn't mistake a marker placement for a click and drag to zoom. If you move the mouse slightly while placing a marker, it won't register it as a click and drag unless you hold down the mouse, and move it, then it waits for either 300ms or a movement of more than 5 pixels. It also highlights the selected region with the highlight color of whatever color theme you're using. I checked it with the default light, dark, high contrast, and some other custom color themes for good measure!
